zcash_client_sqlite/src/wallet.rs

```diff
@@ -1360,15 +1356,15 @@ pub(crate) fn get_derived_account<P: consensus::Parameters>(
          WHERE hd_seed_fingerprint = :hd_seed_fingerprint
          AND hd_account_index = :hd_account_index
          AND (
-             :zcashd_legacy_address_index IS NULL
+             :zcashd_legacy_address_index = -1
```

Comment:
nit: we use `encode/decode`, but this being hardcoded here means that the encoding and decoding are sharing knowledge about the implementation of these queries. Maybe use a constant instead?